tests/Integration/Decoder.elm

```diff
@@ -144,4 +163,7 @@ all =
         , test "should decode Rules" <|
             always <|
                 Expect.equal (Ok expectedRules) (decodeString decodeRules rulesJson)
+        , test "should decode labels" <|
+            always <|
+                Expect.equal (Ok expectedLabels) (decodeString decodeLabels labelsJson)
```

Comment:
Added 3 more tests: empty dict, dict with object, dict with ref. Good idea! I was missing the flattening in my initial implementation. Fixed now.